### PR TITLE
Deprecated FlowLogic.getCounterpartyMarker as it's complicated and probably not used

### PR DESCRIPTION
**Pull request by shamsasari**
The replacement for it is to use sub-flows.

Also made flow registration require the client flow class rather than any old class.

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/flows/FlowLogic.kt [hunk 1]
```diff
@@ -190,9 +188,9 @@ abstract class FlowLogic<out T> {
 
     private var _stateMachine: FlowStateMachine<*>? = null
     /**
-     * Internal only. Reference to the [Fiber] instance that is the top level controller for the entire flow. When
-     * inside a flow this is equivalent to [Strand.currentStrand]. This is public only because it must be accessed
-     * across module boundaries.
+     * Internal only. Reference to the [co.paralleluniverse.fibers.Fiber] instance that is the top level controller for
+     * the entire flow. When inside a flow this is equivalent to [co.paralleluniverse.strands.Strand.currentStrand]. This
+     * is public only because it must be accessed across module boundaries.
```
Comment (mikehearn) [inline core/src/main/kotlin/net/corda/core/flows/FlowLogic.kt]: can you add a `@suppress` inside the api doc (to stop it showing up in javadoc/kdoc)

File: node/src/main/kotlin/net/corda/node/services/statemachine/FlowStateMachineImpl.kt [hunk 2]
```diff
@@ -304,15 +301,26 @@ class FlowStateMachineImpl<R>(override val id: StateMachineRunId,
         logger.trace { "Initiating a new session with $otherParty" }
         val session = FlowSession(sessionFlow, random63BitValue(), null, FlowSessionState.Initiating(otherParty))
         openSessions[Pair(sessionFlow, otherParty)] = session
-        val counterpartyFlow = sessionFlow.getCounterpartyMarker(otherParty).name
-        val sessionInit = SessionInit(session.ourSessionId, counterpartyFlow, firstPayload)
+        sessionFlow.javaClass.superclass
```
Comment (mikehearn) [inline node/src/main/kotlin/net/corda/node/services/statemachine/FlowStateMachineImpl.kt]: What's this dead expression for?

Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/services/statemachine/FlowStateMachineImpl.kt]: Whoops!

**Comment by shamsasari** [comment]
Comments addressed

**Pull request merged**